### PR TITLE
48049 error diagnostic command palette

### LINKED ISSUE #48049: Add Diagnostic Info as command palette action
Can we add "Download diagnostic information" as an action in the command palette?

[Slack Message](https://metaboat.slack.com/archives/C064EB1UE5P/p1726857126043729)

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/48049 

### Description
Adds an option in the command palette to open the Error Diagnostic modal. This involved moving the state of the modal to redux.

### How to verify
1. Press `cmd+k` to open the command palette, and type `Error`
2. You should see the option `Open error diagnostic modal`. Select it
3. You should be presented with the diagnostic modal.
4. close the modal, and then press `cmd+f1` on OSX, or `ctrl+f1` on Windows. The modal should re-open

### Demo
chrome_40ZWc8Taqw

### Checklist
- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts [hunk 1]
```diff
@@ -42,6 +45,25 @@ describe("error reporting modal", () => {
...
 
+  it("should allow you to open the error reporting modal via the command palette", () => {
+    restore();
+    cy.signInAsAdmin();
+    cy.visit("/");
+
+    cy.findByTestId("home-page")
+      .findByText(/see what metabase can do/i)
+      .realClick();
+    cy.wait(500);
```
Comment (sloansparger) [inline e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts:56]: wait bad. at least put a comment on why it is needed.

File: frontend/src/metabase/palette/hooks/useCommandPaletteBasicActions.tsx [hunk 2]
```diff
@@ -186,7 +187,18 @@ export const useCommandPaletteBasicActions = ({
       },
     ];
 
-    return [...actions, ...browseActions];
+    const diagnosticAction = {
+      id: "diagnostic_modal",
+      name: t`Open error diagnostic modal`,
+      section: "basic",
+      icon: "info",
+      shortcut: ["$mod+f1"],
```
Comment (sloansparger) [inline frontend/src/metabase/palette/hooks/useCommandPaletteBasicActions.tsx:195]: this shortcut definition is so much cleaner!

**Review by sloansparger — APPROVED** [review]

**Comment by sloansparger** [comment]
approved, though would like to see some kind of comment around the `cy.wait` if it can't be removed or replaced with something else (e.g. wait on some network request).

File: e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts (hunk 1 above)
Comment (npfitz) [reply e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts]: I was able to remove it, this was just copied from the test above it. Basically you need to give the app time to register listeners before you try to use keyboard shortcuts (I faced something similar when writing command palette tests). I transitioned to opening the command palette via the search button instead

File: frontend/src/metabase/palette/hooks/useCommandPaletteBasicActions.tsx (hunk 2 above)
Comment (npfitz) [reply frontend/src/metabase/palette/hooks/useCommandPaletteBasicActions.tsx:195]: 🥳

**Pull request merged**